src/service.js

```diff
@@ -1,20 +1,39 @@
 let express = require('express');
 let http = require('http');
+let timeout = require('connect-timeout');
 
 let server;
 
 exports.start = function start(port = 8080) {
   let app = express();
+
+  // We shouldn't timeout log requests.
+  app.get('/log', require('./log'));
+
+  app.use(timeout('5s'));
+
   app.get('/', (req, res) => res.send('200 OK'));
   app.get('/crashes', require('./list_crash_reports'));
   app.get('/crashes/:id', require('./get_crash_report'));
   app.get('/info', require('./info'));
-  app.get('/log', require('./log'));
-  app.get('*', (req, res) => res.status(404).send('404 Not Found'));
   app.post('/connection/:port', require('./connect'));
   app.post('/profile', require('./profile'));
   app.post('/restart', require('./restart'));
   app.delete('/connection/:port', require('./disconnect'));
+
+  // Error handling.
+  app.use((err, req, res, next) => {
```

Comment:
With the rebase, and for cleanliness, can we move this into a separate middleware file, e.g. error.js?